src/parser/function.rs

```diff
@@ -1,20 +1,20 @@
-use crate::lexer::Token;
+use crate::lexer::token::Token;
 use crate::parser::ast::{Function, Parameter};
 use crate::parser::error::ParseResult;
 use crate::parser::span::{Spanned, WithSpan};
 use crate::parser::Parser;
 use bumpalo::collections::Vec;
 
-impl<'s, 'a> Parser<'s, 'a> {
-    fn parse_parameter(&mut self) -> ParseResult<Parameter<'s>> {
+impl<'s, 'a, 'e> Parser<'s, 'a, 'e> {
```

Comment:
Why is `'e` added? It's not used in the diff as far as I can tell, but maybe I'm not looking in the right places.